pkg/services/guardian/accesscontrol_guardian.go

```diff
@@ -292,22 +292,22 @@ func (a *accessControlFolderGuardian) CanDelete() (bool, error) {
 	return a.evaluate(accesscontrol.EvalPermission(dashboards.ActionFoldersDelete, dashboards.ScopeFoldersProvider.GetResourceScopeUID(a.folder.UID)))
 }
 
-func (a *accessControlDashboardGuardian) CanCreate(folderID int64, isFolder bool) (bool, error) {
+func (a *accessControlDashboardGuardian) CanCreate(folderUID string, isFolder bool) (bool, error) {
```

Comment:
@IevaVasiljeva would the guardian work for all versions, if we change the folderID from int64 to the folderUID?